### ISSUE TITLE
create puppeteer-core package on npm

### ISSUE THREAD

**Issue by graingert** [puppeteer/puppeteer#903]
Feature request, it would be great to be able to depend on puppeteer in other npm packages, but it automatically downloads Chromium which is a bit overkill for some people.

Turn this repo into a lerna repo with two packages:
```sh
npm install puppeteer-core # gets a pure JS bundle with no install side effects
npm install puppeteer # depends on puppeteer-core and installs chrome
```

**Comment by ebidel** [comment]
I think we're already covered by `PUPPETEER_SKIP_CHROMIUM_DOWNLOAD `. 

https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#environment-variables

**Comment by graingert** [comment]
Yeah this isn't so good for other module authors though

**Comment by ebidel** [comment]
They're available as npm config options too. The project can include an `.npmrc` and use `puppeteer_skip_chromium_download`. Or define the env variable in a `preinstall` script.

**Comment by graingert** [comment]
@ebidel but then that stops it working for every other module

**Comment by ebidel** [comment]
Maybe I'm misunderstanding [per-project configs](https://docs.npmjs.com/files/npmrc#per-project-config-file). I read that as only affecting the current project where you run `npm`.  Does npm install not apply the`.npmrc` of dep packages? 


...I guess if your project sets this config, it may prevent sub packages that depend on pupppteer to not download chromium as well, but I'm not sure that would be very common.

**Comment by graingert** [comment]
@ebidel that works per project, but not per module. Using multiple singe purpose modules is a lot easier to consume for everyone and .npmrc files.

**Comment by graingert** [comment]
providing a core library that doesn't download Chromium fits the goals of this project:

> Provide a **slim**, canonical library

**Comment by LeMoussel** [comment]
@ebidel, "_Or define the env variable in a preinstall script._" I  do this in `package.json`

> {
>   "name": "puppeteertest",
>   "version": "1.0.0",
>   "description": "",
>   "main": "index.js",
>   "scripts": {
>     "preinstall": "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true",
>     "test": "echo \"Error: no test specified\" && exit 1"
>   },
>   "author": "LeMoussel",
>   "license": "ISC"
> }

And when I install `puppeteer`like this, 
> $ npm install --save puppeteer

I got
> puppeteer@0.11.0 install C:\Users\lemoussel\OneDrive\nodejs\puppeteerTest\node_modules\puppeteer
> node install.js
> 
> 
> Chromium downloaded to C:\Users\lemoussel\nodejs\puppeteerTest\node_modules\puppeteer\.local-chromium\win64-499413
> npm notice created a lockfile as package-lock.json. You should commit this file.
> npm WARN puppeteertest@1.0.0 No description
> npm WARN puppeteertest@1.0.0 No repository field.
> 
> + puppeteer@0.11.0
> added 42 packages in 128.337s

Chromium is downloaded during installation step.

**Comment by ebidel** [comment]
@LeMoussel I wouldn't expect that one to work. If you create an `.npmrc` file in your project folder and add `puppeteer_skip_chromium_download = true`, that should prevent the download.

@graingert so to confirm...
> Does npm install not apply the.npmrc of dep packages?

Sounds like no? From my limited testing trying to `npm install` a folder that contains an `.npmrc` with the config var, it doesn't work. That may explain this note in the docs:

> Note that this only applies to the root of the project that you're running npm in. It has no effect when your module is published. 

...we probably do want an easier solution for module authors that want to depend on Puppeteer but not install chromium.

**Comment by graingert** [comment]
Yep eg:

```
puppeteer
|--> @google/puppeteer-core
\--> @google/chromium
```

Then users can install puppeteer or `@google/chromium` and library authors can install `@google/puppeteer-core`

**Comment by aslushnikov** [comment]
There are no plans for this right now; we're focused on implementing core functionality. We might get back to this later as all the main challenges are completed.

**Comment by sinoon** [comment]
`PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true npm i` work for me

**Comment by graingert** [comment]
This is for third party packages

**Comment by doodlecube** [comment]
+1 for puppeteer-core or similar for use in package.json.

appreciate this is closed maybe later revisit, just posting this if helps anyone.

I found solution that works for me in a seed project.
don't know if would work for module developer.

in package.json
"scripts": {
    "preinstall": "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true npm i puppeteer"
  }

or to specify version
"preinstall": "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true npm i puppeteer@0.12"

run "npm install", the preinstall takes care of puppeteer first without downloading chromium.

note.
my npm is set to save any unlisted install to dependencies
(so after first running this puppeteer is then listed in dependencies).
this is not too much of an issue, an inconvenience if you are maintaining version number.
I tested by deleting initially downloaded node_modules and run npm install again with puppeteer also in dependencies. everything installs as it did first time and still missed the chromium download.

also works if you prefer in devDependencies:
"preinstall": "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true npm i puppeteer@0.12 --save-dev"
